### PR TITLE
test: Improve ONNXtests reliability

### PR DESCRIPTION
**Pull request by svotaw**
## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Improve ONNX Hub test reliability

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change any dependencies?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure the dependencies are resolved correctly, and list changes here.

## Does this PR add a new feature? If so, have you added samples on website?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure you have added samples following below steps.

1. Find the corresponding markdown file for your new feature in `website/docs/documentation` folder.
   Make sure you choose the correct class `estimators/transformers` and namespace.
2. Follow the pattern in markdown file and add another section for your new API, including pyspark, scala (and .NET potentially) samples.
3. Make sure the `DocTable` points to correct API link.
4. Navigate to website folder, and run `yarn run start` to make sure the website renders correctly.
5. Don't forget to add `<!--pytest-codeblocks:cont-->` before each python code blocks to enable auto-tests for python samples.
6. Make sure the `WebsiteSamplesTests` job pass in the pipeline.

### CODE REVIEW

File: deep-learning/src/main/scala/com/microsoft/azure/synapse/ml/onnx/ONNXHub.scala [hunk 1]
```diff
@@ -71,28 +72,44 @@ object ONNXHubJsonProtocol extends DefaultJsonProtocol {
...
+  }
+
+  def this(modelCacheDir: Path) = {
+    this(
+      modelCacheDir,
+      ONNXHub.DefaultConnectTimeout,
+      ONNXHub.DefaultReadTimeout,
+      ONNXHub.DefaultRetryCount,
+      ONNXHub.DefaultRetryTimeoutInSeconds)
   }
```
Comment (mhamilton723) [inline deep-learning/src/main/scala/com/microsoft/azure/synapse/ml/onnx/ONNXHub.scala:113]: this is a little bit involved, lets just make this take default args to have a single entry point

**Review by mhamilton723 — CHANGES_REQUESTED** [review]
left 1 comment, thanks for the fix!!

**Review by mhamilton723 — APPROVED** [review]

File: deep-learning/src/main/scala/com/microsoft/azure/synapse/ml/onnx/ONNXHub.scala (hunk 1 above)
Comment (svotaw) [reply deep-learning/src/main/scala/com/microsoft/azure/synapse/ml/onnx/ONNXHub.scala:113]: I agree it's messier than I'd like, but the problem is that the main default arg we want is not a constant (modelCacheDir). if we don't supply the extra constructors, then changing ANY of the other defaults requires the user to supply the cache directory themselves. This seems like something we don't want the user to have to do.

**Pull request merged**